src/shared/firewall-util.h

```diff
@@ -30,3 +31,33 @@ int fw_add_local_dnat(
                 const union in_addr_union *remote,
                 uint16_t remote_port,
                 const union in_addr_union *previous_remote);
+
+struct NFTSetContext {
+        int nfproto;
+        char *table;
+        char *set;
+};
+typedef struct NFTSetContext NFTSetContext;
+
+struct NFTSetContexts {
+        NFTSetContext *ctx;
+        size_t n;
```

Comment:
usually we don't abbreviate field names in structures so heavily (we abbreviate to single character names only in local variables in functions). hence, maybe call these fields `contexts` and `n_contexts`

Reply:
yes, much nicer